### ISSUE TITLE
Evaluation scripts for language-guided generation

### ISSUE THREAD

**Issue by janismdhanbad** [yccyenchicheng/AutoSDF#3]
Hi Authors,

Thank you for your amazing work! I find the results very interesting however I couldn't find evaluation scripts for the language-guided generation. Would it be possible to provide them, or point to the correct source? Thank you!

**Comment by paritoshmittal12** [comment]
Hi Janpreet,

Thanks a lot for taking interest in our work. For the evaluation of
language-guided generation, we leverage the nicely maintained codebase of
ShapeGlot <https://github.com/optas/shapeglot>. They propose a neural
listener which we adopt. Specifically, we change the setting to one target
(our generation) and one distractor (GT or result from baselines). We
finetune the off-the-shelf available VGG and PC-AE for extracting shape
features.

I hope it helps.

**Comment by janismdhanbad** [comment]
Thanks a lot for taking the time to give a detailed reply. I took their `train_listener.ipynb` and used the triplets in shapeglot data as follows: I used the target as the actual shape corresponding to the text and randomly picked one shape from the rest of the two shapes as a distractor. Using the already provided features from PC-AE, I got ~86% accuracy on the test dataset as opposed to ~83% mentioned in the paper. I did not change any other default parameters. Is that reasonable or would you recommend making any other changes

**Comment by paritoshmittal12** [comment]
I think the test accuracy is reasonable. We finetuned a PointCloud AutoEncoder (PC-AE), VGG, and then used those features to train the listener. This change in used features can result in different accuracy numbers. 

Yes, during test time we use different networks as targets and distractors. For P(Conf) we assume that if |P(Model a) - P(Model b)| < 0.2 then listener is confused. Please refer to the paper for more details. 

As an aside, we hopefully plan to release the evaluation script for our work within the next few weeks.

**Comment by janismdhanbad** [comment]
Hi Paritosh,
Thanks for the reply, all that makes sense to me. Looking forward to the code release of the evaluation script :)

**Comment by youngstu** [comment]
> As an aside, we hopefully plan to release the evaluation script for our work within the next few weeks.

Could you provide the language-guided evaluation script for subsequent work metric alignment and referencing. 

Very Thanks.

**Comment by youngstu** [comment]
> Hi Paritosh, Thanks for the reply, all that makes sense to me. Looking forward to the code release of the evaluation script :)

Could you provide the language-guided evaluation script for subsequent work metric alignment and referencing.

Very Thanks.
